src/plugins/intel_cpu/src/nodes/non_max_suppression.cpp

```diff
@@ -631,52 +118,82 @@ void NonMaxSuppression::initSupportedPrimitiveDescriptors() {
 
-    checkOutput(getOutputShapeAtPort(NMS_SELECTEDINDICES), supportedIntOutputPrecision, "selected_indices", NMS_SELECTEDINDICES);
-    checkOutput(getOutputShapeAtPort(NMS_SELECTEDSCORES), supportedFloatPrecision, "selected_scores", NMS_SELECTEDSCORES);
+    checkOutput(getOutputShapeAtPort(NMS_SELECTED_INDICES), supportedIntOutputPrecision, "selected_indices", NMS_SELECTED_INDICES);
+    checkOutput(getOutputShapeAtPort(NMS_SELECTED_SCORES), supportedFloatPrecision, "selected_scores", NMS_SELECTED_SCORES);
 
     std::vector<PortConfigurator> inDataConf;
     inDataConf.reserve(inputShapes.size());
     for (size_t i = 0; i < inputShapes.size(); ++i) {
-        Precision inPrecision = i == NMS_MAXOUTPUTBOXESPERCLASS ? Precision::I32 : Precision::FP32;
+        Precision inPrecision = i == NMS_MAX_OUTPUT_BOXES_PER_CLASS ? Precision::I32 : Precision::FP32;
         inDataConf.emplace_back(LayoutType::ncsp, inPrecision);
     }
 
     std::vector<PortConfigurator> outDataConf;
     outDataConf.reserve(outputShapes.size());
     for (size_t i = 0; i < outputShapes.size(); ++i) {
-        Precision outPrecision = i == NMS_SELECTEDSCORES ? Precision::FP32 : Precision::I32;
+        Precision outPrecision = i == NMS_SELECTED_SCORES ? Precision::FP32 : Precision::I32;
         outDataConf.emplace_back(LayoutType::ncsp, outPrecision);
     }
 
-    impl_desc_type impl_type;
-    if (mayiuse(cpu::x64::avx512_core)) {
-        impl_type = impl_desc_type::jit_avx512;
-    } else if (mayiuse(cpu::x64::avx2)) {
-        impl_type = impl_desc_type::jit_avx2;
-    } else if (mayiuse(cpu::x64::sse41)) {
-        impl_type = impl_desc_type::jit_sse42;
-    } else {
-        impl_type = impl_desc_type::ref;
+    impl_desc_type impl_type = impl_desc_type::ref;
+
+#if defined(OPENVINO_ARCH_X86_64)
+    using namespace dnnl::impl::cpu;
+
+    // As only FP32 and ncsp is supported, and kernel is shape agnostic, we can create here. There is no need to recompilation.
+    createJitKernel();
+
+    x64::cpu_isa_t actual_isa = x64::isa_undef;
+    if (m_jit_kernel) {
+        actual_isa = m_jit_kernel->getIsa();
+    }
+    switch (actual_isa) {
+        case x64::avx512_core: impl_type = impl_desc_type::jit_avx512; break;
+        case x64::avx2:        impl_type = impl_desc_type::jit_avx2;   break;
+        case x64::sse41:       impl_type = impl_desc_type::jit_sse42;  break;
+        default:               impl_type = impl_desc_type::ref;
     }
+#endif // OPENVINO_ARCH_X86_64
 
     addSupportedPrimDesc(inDataConf, outDataConf, impl_type);
+}
 
-    // as only FP32 and ncsp is supported, and kernel is shape agnostic, we can create here. There is no need to recompilation.
-    createJitKernel();
+void NonMaxSuppression::createPrimitive() {
+    const auto inputs_num = inputShapes.size();
+
+    if (inputs_num > NMS_MAX_OUTPUT_BOXES_PER_CLASS && m_const_inputs[NMS_MAX_OUTPUT_BOXES_PER_CLASS]) {
```

Comment:
I cannot understand why we need these params initialization on compilation stage. They are requered only on runtime stage and can be loaded from memory inside execute() (esp given we already have such logic there).
Beside that we report only FP32 precision inside `addSupportedPrimDesc` for these inputs -> getScalar() is not required

Reply:
This was done in order to improve performance.
Agree about precisions, but prefer to leave fn getScalar() to support f16 or i64 (for indices) in future.